sklearn/metrics/_ranking.py

```diff
@@ -221,17 +229,27 @@ def _binary_uninterpolated_average_precision(
         return -np.sum(np.diff(recall) * np.array(precision)[:-1])
 
     y_type = type_of_target(y_true, input_name="y_true")
-    if y_type == "multilabel-indicator" and pos_label != 1:
+
+    # Convert to Python primitive type to avoid NumPy type / Python str
+    # comparison. See https://github.com/numpy/numpy/issues/6784
+    present_labels = np.unique(y_true).tolist()
+
+    if y_type == "binary":
+        if len(present_labels) == 2 and pos_label not in present_labels:
+            raise ValueError(
+                f"pos_label={pos_label} is not a valid label. It should be "
+                f"one of {present_labels}"
+            )
+
+    elif y_type == "multilabel-indicator" and pos_label != 1:
         raise ValueError(
-            "Parameter pos_label is fixed to 1 for "
-            "multilabel-indicator y_true. Do not set "
-            "pos_label or set pos_label to 1."
+            "Parameter pos_label is fixed to 1 for multilabel-indicator y_true. "
+            "Do not set pos_label or set pos_label to 1."
         )
-    elif y_type == "binary":
-        # Convert to Python primitive type to avoid NumPy type / Python str
-        # comparison. See https://github.com/numpy/numpy/issues/6784
-        present_labels = np.unique(y_true).tolist()
-        if len(present_labels) == 2 and pos_label not in present_labels:
+
+    elif y_type == "multiclass":
+        y_true = label_binarize(y_true, classes=present_labels)
+        if pos_label not in present_labels:
```

Comment:
I do not think `pos_label` makes sense for `multiclass`. For example, using `pos_label=2` would give zero:

```python
import numpy as np
from sklearn.metrics import average_precision_score
y_true = np.array([0, 0, 1, 1, 2, 2])
y_scores = np.array([
    [0.7, 0.2, 0.1],
    [0.4, 0.3, 0.3],
    [0.1, 0.8, 0.1],
    [0.2, 0.3, 0.5],
    [0.4, 0.4, 0.2],
    [0.1, 0.2, 0.7],
])

# This gives warnings about no positive class
average_precision_score(y_true, y_scores, pos_label=)
0.0
```

I prefer to error if `pos_label != 1`, like the multilabel-indicator case.

@glemaitre WDYT?

Reply:
Indeed, `pos_label` is only meaningful for the binary classification case.
I think this is reasonable to raise an error.

Reply:
Yes, `pos_label` only makes sense for the binary case. I will add the error for the `pos_label != 1` case.